client/components/SnackBar.tsx

```diff
@@ -0,0 +1,27 @@
+import { Button } from '@mui/material';
+import { useSnackbar, OptionsWithExtraProps } from 'notistack';
+import React from 'react';
+
+export function useAlert() {
+  const { enqueueSnackbar, closeSnackbar } = useSnackbar();
+  const action = (snackbarId: string) => (
+    <>
+      <Button onClick={() => {
+        closeSnackbar(snackbarId);
+      }}>
+        Dismiss
+      </Button>
+    </>
+  );
+
+  const successOptions: OptionsWithExtraProps<'success'> = { action, autoHideDuration: 6000, variant: 'success' };
+  const errorOptions: OptionsWithExtraProps<'error'> = { action, autoHideDuration: 6000, variant: 'error' };
+  return (
+    { showSuccess: (message: string, props?: OptionsWithExtraProps<'success'>) => {
+      enqueueSnackbar(message, { ...successOptions, ...props });
+    },
+    showError: (message: string, props?: OptionsWithExtraProps<'error'>) => {
+      enqueueSnackbar(message, { ...errorOptions, ...props });
+    } }
+  );
```

Comment:
nit: styling - better for return values to be as simple as possible to make it really obvious what's being returned from this hook.

```suggestion
  const showSuccess = (message: string, props?: OptionsWithExtraProps<'success'>) => {
    enqueueSnackbar(message, { ...successOptions, ...props });
  };
  const showError = (message: string, props?: OptionsWithExtraProps<'error'>) => {
    enqueueSnackbar(message, { ...errorOptions, ...props });
  };
  
  return {
    showSuccess,
    showError,
  };
```